### PR TITLE
{lang}[foss/2021b] R v4.1.2

### PR DESCRIPTION
**Pull request by branfosj**
(created using `eb --new-pr`)

replaces #14054, as R 4.1.2 is out.

* I've added in ncdf4, rather than having it as a separate ec - all the dependencies are already here so I see no reason for it to be a separate ec

### CODE REVIEW

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bb-pg-login02.bear.cluster - Linux RHEL 8.3, x86_64, Intel(R) Xeon(R) CPU E5-2620 v4 @ 2.10GHz (broadwell), Python 3.6.8
See https://gist.github.com/9049d4e106a7d225e80782471cd54b22 for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
bear-pg0211u03a.bear.cluster - Linux RHEL 8.3, x86_64, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz (cascadelake), Python 3.6.8
See https://gist.github.com/10fde44ae5f87cd23f21e924720b1ae4 for a full test report.

File: easybuild/easyconfigs/r/R/R-4.1.2-foss-2021b.eb [hunk 1]
```diff
@@ -0,0 +1,3080 @@
...
+    'source_urls': [
+        'https://cran.r-project.org/src/contrib/Archive/%(name)s',  # package archive
+        'https://cran.r-project.org/src/contrib/',  # current version of packages
+        'https://cran.freestatistics.org/src/contrib',  # mirror alternative for current packages
+    ],
+    'source_tmpl': '%(name)s_%(version)s.tar.gz',
+}
+
+# !! order of packages is important !!
+# packages updated on 2nd November 2021
```
Comment (verdurin) [inline easybuild/easyconfigs/r/R/R-4.1.2-foss-2021b.eb:80]: Is this a bit of optimism...?

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0211u03a.bear.cluster - Linux RHEL 8.3, x86_64, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz (cascadelake), Python 3.6.8
See https://gist.github.com/cc9b92c8c9f3ccaf07a25cf0cdfb59d7 for a full test report.

File: easybuild/easyconfigs/r/R/R-4.1.2-foss-2021b.eb (hunk 1 above)
Comment (branfosj) [reply easybuild/easyconfigs/r/R/R-4.1.2-foss-2021b.eb]: fixed in de56345

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 3 out of 3 (1 easyconfigs in total)
easybuild-c7.novalocal - Linux centos linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 3.6.8
See https://gist.github.com/b6fd41a273aa43f8fefcac9ee7ed5769 for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0104u09b.bear.cluster - Linux RHEL 8.4, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/834cef2cebe75981fd09b2f0c3d40bb5 for a full test report.

**Comment by verdurin** [comment]
Inevitable checksum fun:
```
== 2021-11-02 07:10:49,382 filetools.py:1301 DEBUG Computed sha256 checksum for /mnt/ds918plus/eb/common/sources/r/R/R-4.1.2.tar.gz: e62643d2f66588e2c745c9f6c0b5559c60db5a71b0584b1ece31dd2a8179eb21 (correct checksum: 2036225e9f7207d4ce097e54972aecdaa8b40d7d9911cd26491fac5a0fab38af)
== 2021-11-02 07:10:49,388 build_log.py:169 ERROR EasyBuild crashed with an error (at easybuild/base/exceptions.py:124 in __init__): Checksum verification for /mnt/ds918plus/eb/common/sources/r/R/R-4.1.2.tar.gz using 2036225e9f7207d4ce097e54972aecdaa8b40d7d9911cd26491fac5a0fab38af failed. (at easybuild/framework/easyblock.py:2211 in checksum_step)
```

**Comment by branfosj** [comment]
```
$ wget https://cran.r-project.org/src/base/R-4/R-4.1.2.tar.gz
$ sha256sum R-4.1.2.tar.gz
2036225e9f7207d4ce097e54972aecdaa8b40d7d9911cd26491fac5a0fab38af  R-4.1.2.tar.gz
$ rm R-4.1.2.tar.gz
$ wget https://cloud.r-project.org/src/base/R-4/R-4.1.2.tar.gz
$ sha256sum R-4.1.2.tar.gz
2036225e9f7207d4ce097e54972aecdaa8b40d7d9911cd26491fac5a0fab38af  R-4.1.2.tar.gz
```

I'm consistently getting the `203...` cehcksums that is in the easyconfig. @verdurin can you check whether there is an issue with your download?

**Comment by verdurin** [comment]
@branfosj yes, I think that's a false alarm. Running a second build now.

**Comment by branfosj** [comment]
@boegelbot please test @ generoso
EB_ARGS="--parallel-extensions-install --experimental"
CORE_CNT=16

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
centos7.lan - Linux centos linux 7.9.2009, x86_64, Intel Core Processor (Skylake, IBRS), Python 3.6.8
See https://gist.github.com/ad3488172f6b4cb5eb9f832ffca83714 for a full test report.

**Review by verdurin — APPROVED** [review]
Looks fine.

**Comment by verdurin** [comment]
Going in, thanks @branfosj!

**Pull request merged**